### ISSUE TITLE
Plus across DST boundary not working as expected

### ISSUE THREAD

**Issue by tjk** [moment/luxon#648]
```javascript
luxon.DateTime
  .local(2016, 10, 30, 3, 59)
  .setZone("Europe/Athens", {keepLocalTime: true})
  .plus({minutes: 1})
  .hour
//=> 4
```

> Daylight saving time 2016 in Greece began at 3:00 AM on
> Sunday, March 27
> and ended at 4:00 AM on
> Sunday, October 30
> All times are in Eastern European Time.

I expected the hour to be 3 -- am I doing something wrong? Thanks for any pointers.

PS: Was porting https://github.com/harrisiirak/cron-parser from `moment-timezone` to `luxon` (and this is is one of the tests that's failing: https://github.com/harrisiirak/cron-parser/blob/master/test/timezone.js#L149)

**Comment by icambron** [comment]
Luxon has undefined behavior for which of the two 3:59s it will pick. My guess offhand is that something about the way setZone works caused it to pick a different one that you expected.

Here's the relevant section of the docs: https://moment.github.io/luxon/docs/manual/zones.html#ambiguous-times.

**Comment by tjk** [comment]
In that case, shouldn't the following return 3 since 2:59 is not ambiguous?

```
luxon.DateTime
  .local(2016, 10, 30, 2, 59)
  .setZone("Europe/Athens", {keepLocalTime: true})
  .plus({minutes: 61})
  .hour
//=> 4
```

**Comment by icambron** [comment]
Hmm, yeah, this looks like a bug in `keepLocalTime`: 

```js
luxon.DateTime
  .local(2016, 10, 30, 2, 59)
  .setZone("Europe/Athens", {keepLocalTime: true})
  .toISO()
"2016-10-30T03:59:00.000+03:00"
```

A workaround:

```js
luxon.DateTime
  .fromObject({ year: 2016, month: 10, day: 30, hour: 2, minute: 59, zone: "Europe/Athens" })
  .toISO()
"2016-10-30T02:59:00.000+03:00"
```

I'll have to dig into why `keepLocalTime` is being weird there

**Comment by tjk** [comment]
Thanks for that workaround!

**Comment by icambron** [comment]
In v2, `local()` will take a `zone` option, so you won't have to use `setZone`. It should still work though...

**Comment by icambron** [comment]
Fixed in d4fd0af, will release shortly
